bundles/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/IhcBinding.java

```diff
@@ -396,28 +399,79 @@ private void updateResource(String itemName, Type type, boolean updateOnlyOutBin
 			
 			try {
 
-				int resourceId = provider.getResourceId(itemName);
-				WSResourceValue value = ihc
-						.getResourceValueInformation(resourceId);
+				int resourceId = provider.getResourceId(itemName, (Command)type);
+				logger.trace(
+						"found resourceId {} (item='{}', state='{}', class='{}')",
+						new Object[] { new Integer(resourceId).toString(), itemName, type.toString(),
+								type.getClass().toString() });
+				if(resourceId > 0) {
 				
-				ArrayList<IhcEnumValue> enumValues = null;
-
-				if (value instanceof WSEnumValue) {
-
-					enumValues = ihc.getEnumValues(((WSEnumValue) value)
-							.getDefinitionTypeID());
-				}
-
-				value = IhcDataConverter.convertCommandToResourceValue(type, value,
-						enumValues);
+					WSResourceValue value = ihc
+							.getResourceValueInformation(resourceId);
+					
+					ArrayList<IhcEnumValue> enumValues = null;
+	
+					if (value instanceof WSEnumValue) {
+	
+						enumValues = ihc.getEnumValues(((WSEnumValue) value)
+								.getDefinitionTypeID());
+					}
+	
+					Integer val = provider.getValue(itemName, (Command)type);
```

Comment:
Could you add comment here, how this val and trigger special case works. Without comment, code could be little bit confusing  afterwards.